arches/app/frontend/webpack/webpack.common.js

```diff
@@ -0,0 +1,213 @@
+const Path = require('path');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const BundleTracker = require('webpack-bundle-tracker');
+const webpack = require('webpack');
+const fs = require('fs');
+
+const buildJavascriptFilepathLookup = function(path, outerAcc) {
+    return fs.readdirSync(path).reduce((acc, name) => {
+        if (fs.lstatSync(path + '/' + name).isDirectory() ) {
+            return buildJavascriptFilepathLookup(path + '/' + name, acc);
+        }
+        else {
+            const subPath = (path + '/' + name).slice(12);
+            const subPathWithoutFileExtension = subPath.slice(0, subPath.length - 3);
+
+            return { 
+                ...acc, 
+                [subPathWithoutFileExtension]: { 'import': `${path}/${name}`, 'filename': `js/[name].js` } 
+            };
+        }
+    }, outerAcc);
+};
+
+const entryPointConfiguration = buildJavascriptFilepathLookup('../media/js', {});
+
+var javascriptRelativeFilepathToAbsoluteFilepathLookup = Object.keys(entryPointConfiguration).reduce((acc, path) => {
+    acc[path + '$'] = Path.resolve(__dirname, `../../media/js/${path}`);
+    return acc;
+}, {});
+
+const buildHtmlWebpackPluginList = function(path, outerAcc) {
+    return fs.readdirSync(path).reduce((acc, name) => {
+        if (fs.lstatSync(path + '/' + name).isDirectory() ) {
+            return buildHtmlWebpackPluginList(path + '/' + name, acc);
+        }
+        else {
+            const subPath = (path + '/' + name).slice(13);
+            const subPathWithoutFileExtension = subPath.slice(0, subPath.length - 4);
+
+            acc.push(new HtmlWebpackPlugin({
+                template: `../templates/${subPath}`, // relative path to the HTML files
+                filename: `templates/${subPath}`, // output HTML files
+                chunks:  [ Path.resolve(__dirname, `../../media/build/js/${subPathWithoutFileExtension}.js`) ] // respective JS files
+            }));
+            return acc;
+        }
+    }, outerAcc);
+};
+
+module.exports = {
+    entry: entryPointConfiguration,
+    output: {
+        path: Path.resolve(__dirname, '../../media/build'),
+        publicPath: '/foo/',
+        libraryTarget: 'amd-require',
+        clean: true,
+    },
+    optimization: {
+        // concatenateModules: true,
+        // flagIncludedChunks: true,
+        // removeAvailableModules: true,
+        // splitChunks: {  //  CURRENTLY DEFAULT VALUES
```

Comment:
all optimization is currently disabled until we get the app working. `splitChunks` config is currently representing the default application config